ext/puma_http11/org/jruby/puma/MiniSSL.java

```diff
@@ -196,6 +192,61 @@ public IRubyObject initialize(ThreadContext threadContext, IRubyObject miniSSLCo
     return this;
   }
 
+  /**
+   * Try to create a netty tcnative OpenSSL engine which has much better performance
+   * than the JDK native one.
+   * Fall back to the default engine otherwise.
+   */
+  private static SSLEngine tryCreateEngine(KeyManagerFactory keyManagerFactory,
+                                               TrustManagerFactory trustManagerFactory,
+                                               String[] protocols,
+                                               long verify_mode,
+                                               String[] ciphers) throws NoSuchAlgorithmException, KeyManagementException {
+    try {
+      io.netty.handler.ssl.ClientAuth clientAuth = io.netty.handler.ssl.ClientAuth.NONE;
```

Comment:
If Netty libraries are not available this will throw a _ClassNotFoundException_ and default back to the original implementation.

Reply:
Does this work? In CI, the build simply fails https://travis-ci.org/github/puma/puma/jobs/661773151#L326

Reply:
Looks like the JARs were not included in the PR; I will add them.